prov/acmp/src/acmp.c

```diff
@@ -1774,6 +1845,7 @@ acmp_resolve_dest(struct acmp_ep *ep, struct acm_msg *msg, uint64_t id)
 		acm_log(2, "have address, resolving route\n");
 		acm_increment_counter(ACM_CNTR_ADDR_CACHE);
 		atomic_inc(&ep->counters[ACM_CNTR_ADDR_CACHE]);
+		acmp_acquire_sgid(saddr, dest);
```

Comment:
For some reason it seems my comment was deleted, sorry about that, I'll write again.

> How does this change handle something like: src=10.0.0.1 dst = 10.0.1.2 followed by src=10.0.0.2 dst=10.0.1.2?

How is it done today?
in acm_svr_resolve_dest we use acm_svr_select_src to select the src addr
then we use  acm_get_ep_address (with the saddr) to find the acmc_addr
and then we jump into acmp

Which means different src address still work, for example:
```
markb@r-freeb14:~ $ ib_acme -V -d 1.1.2.13 -s 1.1.1.144
Service: localhost
Destination: 1.1.2.13
Source: 1.1.1.144
Path information
  dgid: fec0::2:14:500:0:19
  sgid: fec0::1:7cfe:9003:72:6f33
  dlid: 1
  slid: 2
  flow label: 0x0
  hop limit: 255
  tclass: 0
  reversible: 1
  pkey: 0xffff
  sl: 0
  mtu: 5
  rate: 12
  packet lifetime: 18

return status 0x0
markb@r-freeb14:~ $ ib_acme -V -d 1.1.2.13 -s 1.1.1.14
Service: localhost
Destination: 1.1.2.13
Source: 1.1.1.14
Path information
  dgid: fec0::2:14:500:0:19
  sgid: fec0::1:7cfe:9003:72:6f32
  dlid: 1
  slid: 3
  flow label: 0x0
  hop limit: 255
  tclass: 0
  reversible: 1
  pkey: 0xffff
  sl: 0
  mtu: 5
  rate: 12
  packet lifetime: 18

return status 0x0
```

I have a host file without a route file.
Which means I have an ip->gid mapping and that's it.

Now If I run ibacm before ipoib is loaded (and given an ip), I don't have any route rules
to be able to select the SGID based on the dst IP.